WordPress/src/main/java/org/wordpress/android/util/config/AppConfig.kt

```diff
@@ -37,17 +48,42 @@ class AppConfig
      * @param feature feature which we're checking remotely
      */
     fun isEnabled(feature: FeatureConfig): Boolean {
-        if (manualFeatureConfig.hasManualSetup(feature)) {
-            return manualFeatureConfig.isManuallyEnabled(feature)
-        }
-        if (feature.remoteField == null) {
-            return feature.buildConfigValue
+        return featureState(feature).isEnabled
+    }
+
+    /**
+     * Get the enabled flag and the source where it came from.
+     * @param feature feature we're checking remotely
+     */
+    fun featureState(feature: FeatureConfig): FeatureState {
+        return buildFeatureState(feature).also { state ->
+            feature.remoteField?.let {
+                analyticsTracker.track(
+                        FEATURE_FLAG_SET,
+                        feature.remoteField,
+                        state
+                )
+            }
         }
-        return enabledFeatures.getOrPut(feature.remoteField) {
-            val loadedValue = feature.buildConfigValue || remoteConfig.isEnabled(feature.remoteField)
-            enabledFeatures[feature.remoteField] = loadedValue
-            analyticsTracker.track(Stat.FEATURE_FLAG_SET, mapOf(feature.remoteField to loadedValue))
-            loadedValue
+    }
+
+    private fun buildFeatureState(feature: FeatureConfig): FeatureState {
+        return when {
+            manualFeatureConfig.hasManualSetup(feature) -> {
+                ManuallyOverriden(manualFeatureConfig.isManuallyEnabled(feature))
+            }
+            feature.remoteField == null -> {
+                BuildConfigValue(feature.buildConfigValue)
+            }
+            feature.buildConfigValue -> {
+                BuildConfigValue(feature.buildConfigValue)
+            }
+            !remoteConfig.isInitialized(feature.remoteField) -> {
+                DefaultValue(remoteConfig.isEnabled(feature.remoteField))
+            }
+            else -> {
+                RemoteValue(remoteConfig.isEnabled(feature.remoteField))
+            }
```

Comment:
Maybe instead of relying on `RemoteConfig::isInitialized` we could take advantage of [`FirebaseRemoteConfigValue::getSource`](https://firebase.google.com/docs/reference/android/com/google/firebase/remoteconfig/FirebaseRemoteConfigValue?authuser=1#public-abstract-int-getsource)? It looks like it would work really well for what we want to achieve here.

Reply:
this is a good idea 👍 , I haven't notice that's available (but I was wondering why it isn't). I've implemented it (and removed the appPrefs usage) in 22659e4